### PR TITLE
TF-611 Create a new filter email rule

### PR DESCRIPTION
**Pull request by dab246**
#611 

### Resolved

https://user-images.githubusercontent.com/80730648/183887463-5d2332e3-5938-4bd9-9d47-9465c0a8a144.mov

### CODE REVIEW

File: lib/main/localizations/app_localizations.dart [hunk 1]
```diff
@@ -1840,4 +1840,97 @@ class AppLocalizations {
...
+
+  String get notExactlyEquals {
+    return Intl.message(
+        'Not exactly equals',
+        name: 'notExactlyEquals');
+  }
+
+  String get actionTitleRulesFilter {
+    return Intl.message(
+        'Perform the following actions:',
```
Comment (hoangdat) [inline lib/main/localizations/app_localizations.dart:1900]: `*action`

File: lib/features/manage_account/domain/model/create_new_rule_filter_request.dart [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+import 'package:equatable/equatable.dart';
+import 'package:rule_filter/rule_filter/tmail_rule.dart';
+
+class CreateNewRuleFilterRequest with EquatableMixin {
+
+  final List<TMailRule> newListRules;
```
Comment (hoangdat) [inline lib/features/manage_account/domain/model/create_new_rule_filter_request.dart:7]: please separate `current list` and new item will be added

File: lib/features/manage_account/data/network/manage_account_api.dart [hunk 3]
```diff
@@ -142,4 +146,40 @@ class ManageAccountAPI {
 
     return result?.list.first.rules ?? <TMailRule>[];
   }
+
+  Future<List<TMailRule>> createNewEmailRuleFilter(
```
Comment (hoangdat) [inline lib/features/manage_account/data/network/manage_account_api.dart:150]: use the same API from PR: #824

File: lib/features/manage_account/data/datasource_impl/manage_account_datasource_impl.dart [hunk 4]
```diff
@@ -71,4 +72,16 @@ class ManageAccountDataSourceImpl extends ManageAccountDataSource {
       throw error;
     });
   }
+
+  @override
+  Future<List<TMailRule>> createNewEmailRuleFilter(
```
Comment (hoangdat) [inline lib/features/manage_account/data/datasource_impl/manage_account_datasource_impl.dart:77]: 👍

File: lib/features/mailbox/presentation/model/mailbox_actions.dart [hunk 5]
```diff
@@ -11,6 +11,7 @@ enum MailboxActions {
   rename,
   move,
   markAsRead,
+  selectFromRuleFilter,
```
Comment (hoangdat) [inline lib/features/mailbox/presentation/model/mailbox_actions.dart:14]: `selectForRuleAction`

File: lib/main/localizations/app_localizations.dart (hunk 1 above)
Comment (dab246) [reply lib/main/localizations/app_localizations.dart:1900]: what do you mean?

File: lib/features/manage_account/domain/model/create_new_rule_filter_request.dart (hunk 2 above)
Comment (dab246) [reply lib/features/manage_account/domain/model/create_new_rule_filter_request.dart:7]: Why do that?

File: lib/features/manage_account/data/network/manage_account_api.dart (hunk 3 above)
Comment (dab246) [reply lib/features/manage_account/data/network/manage_account_api.dart:150]: IMO, we should separate it so it's easy to change if `jmap` supports `create\delete` later. What do you mean?

File: lib/features/manage_account/domain/model/create_new_rule_filter_request.dart (hunk 2 above)
Comment (dab246) [reply lib/features/manage_account/domain/model/create_new_rule_filter_request.dart]: Ok, to make it clear we will do so. I have updated it.

File: lib/features/manage_account/data/network/manage_account_api.dart (hunk 3 above)
Comment (dab246) [reply lib/features/manage_account/data/network/manage_account_api.dart]: Ok, now we should sync all of them, if there are changes later, we can update the data layer again

File: lib/main/localizations/app_localizations.dart (hunk 1 above)
Comment (hoangdat) [reply lib/main/localizations/app_localizations.dart:1900]: replace `actions` by `action` :)

File: lib/features/manage_account/data/extensions/list_tmail_rule_extensions.dart [hunk 6]
```diff
@@ -0,0 +1,16 @@
+
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:rule_filter/rule_filter/rule_id.dart';
+import 'package:rule_filter/rule_filter/tmail_rule.dart';
+
+extension ListTMailRuleExtensions on List<TMailRule> {
+
+  List<TMailRule> get withId {
```
Comment (hoangdat) [inline lib/features/manage_account/data/extensions/list_tmail_rule_extensions.dart:8]: `addIds` or `withIds`. WDYT?

Comment (dab246) [reply lib/features/manage_account/data/extensions/list_tmail_rule_extensions.dart:8]: oke. sounds reasonable

File: lib/main/localizations/app_localizations.dart (hunk 1 above)
Comment (dab246) [reply lib/main/localizations/app_localizations.dart:1900]: ok

**Review by hoangdat — APPROVED** [review]

**Review by ManhNTX — APPROVED** [review]

**Pull request merged**